File: Find Three Largest Number/O(n)_time_and_O(1)_space_1.py
def findThreeLargestNumbers(array):
    # Write your code here.
    ret=[None,None,None]
    for num  in array:
        updateLargest(num,ret)
    return ret

def updateLargest(num,ret):
    if ret[2] is None or num>ret[2]:
        shiftAndUpdate(ret,num,2)
    elif ret[1] is None or num>ret[1]:
        shiftAndUpdate(ret,num,1)
    elif ret[0] is None or num>ret[0]:
        shiftAndUpdate(ret,num,0)
    
def shiftAndUpdate(ret,num,ind):
    for  i in range(ind+1):
        if ind == i:
            ret[i]=num
        else:
            ret[i]=ret[i+1]

File: Searching/Find Three Largest Number/O(n)_time_and_O(1)_space_1.js
function findThreeLargestNumbers(array) {
    // Write your code here.
    let ret=[null,null,null]
    for (let val of array)
      updateLargest(ret,val)
    return ret
  }
  
  function updateLargest(ret,val){
    if (!ret[2] || val>ret[2]){
      shiftAndUpdate(ret,val,2)
    }else  if (!ret[1] || val>ret[1]){
      shiftAndUpdate(ret,val,1)
    } else  if (!ret[0] || val>ret[0]){
      shiftAndUpdate(ret,val,0)
    }
  }
  
  function shiftAndUpdate(ret,val,ind){
    for (let i=0;i<ind+1;i++){
      if(ind==i){
        ret[i]=val
      }else{
        ret[i]=ret[i+1]
      }
    }
  }
  